src/components/Board/Board.tsx

```diff
@@ -1,125 +1,129 @@
+      }
     }
-
-    return (
-        <>
-            <style>
-                {`.board { --board__columns: ${columnsCount} }`}
-            </style>
-
-            {showPreviousButton && (
-                <button
-                    className={`board__navigation board__navigation-prev ${getColorClassName(columnColors[previousColumnIndex])}`}
-                    onClick={handlePreviousClick}
-                    aria-hidden={true}
-                >
-                    <LeftArrowIcon className="board__navigation-arrow board__navigation-arrow-prev"/>
-                </button>
-            )}
-
-            <main className="board" ref={boardRef}>
-                <div className={`board__spacer-left ${getColorClassName(columnColors[0])}`} />
-                    {children}
-                <div className={`board__spacer-right ${getColorClassName(columnColors[columnColors.length - 1])}`} />
-            </main>
-
-            {showNextButton && (
-                <button
-                    className={`board__navigation board__navigation-next ${getColorClassName(columnColors[(lastVisibleColumnIndex + 1) % columnColors.length])}`}
-                    onClick={handleNextClick}
-                    aria-hidden={true}>
-                    <RightArrowIcon className="board__navigation-arrow board__navigation-arrow-next"/>
-                </button>
-            )}
-        </>
-    )
+    return;
+  }, [children]);
+
+  const columnsCount = React.Children.count(children);
+  if (!children || columnsCount === 0) {
+    return <div className="board--empty">Empty board</div>;
+  }
+
+  const { firstVisibleColumnIndex, lastVisibleColumnIndex } = state;
+  const columnColors = React.Children.map(children, (child) => child.props.color);
+
+  const showNextButton = lastVisibleColumnIndex < columnsCount - 1;
+  const showPreviousButton = firstVisibleColumnIndex > 0;
+
+  const previousColumnIndex = firstVisibleColumnIndex > 0 ? firstVisibleColumnIndex - 1 : columnColors.length - 1;
+  const nextColumnIndex = lastVisibleColumnIndex === columnsCount - 1 ? 0 : firstVisibleColumnIndex + 1;
+
+  const handlePreviousClick = () => {
+    boardRef.current!.children[previousColumnIndex + 1].scrollIntoView({ inline: 'start', behavior: 'smooth' });
+  }
+
+  const handleNextClick = () => {
+    boardRef.current!.children[nextColumnIndex + 1].scrollIntoView({ inline: 'start', behavior: 'smooth' });
+  }
+
+  return (
+    <>
+      <style>
+        {`.board { --board__columns: ${columnsCount} }`}
+      </style>
+
+      {showPreviousButton && (
+        <button
+          className={`board__navigation board__navigation-prev ${getColorClassName(columnColors[previousColumnIndex])}`}
+          onClick={handlePreviousClick}
+          aria-hidden={true}
+        >
+          <LeftArrowIcon className="board__navigation-arrow board__navigation-arrow-prev" />
+        </button>
+      )}
+
+      <BoardUsers numOfUsersToShow={NUM_OF_DISPLAYED_USERS} />
```

Comment:
I think I would prefer to let the `BoardUsers` component deal with the question how many users should be displayed, as it is not related to the board.

Also I think the magical number `4` is only suitable for larger displays and should be set to a smaller number on mobile screens.

Reply:
done